docs/source/tutorial/convnet.rst

```diff
@@ -0,0 +1,428 @@
+list ``net`` easily. :class:`~chainer.Function` doesn't have any trainable
+parameters, so that we can't register it to the model with
+:meth:`~chainer.Chain.add_link`, but we want to use
+:class:`~chainer.Function` s for constructing a forward path. The list
+``net`` is stored as an attribute attr:`forward` to refer it in
+:meth:`__call__`. In :meth:`__call__`, it retrieves all layers in the network
+from :attr:`self.forward` sequentially regardless of what types of object (
+:class:`~chainer.Link` or :class:`~chainer.Function`) it is, and gives the
+input variable or the intermediate output from the previous layer to the
+current layer. The last part of the :meth:`__call__` to switch its behavior
+by the training/inference mode is the same as the former way.
+
+Ways to calculate loss
+......................
+
+When you train the model with label vector ``t``, the loss should be calculated
+using the output from the model. There also are several ways to calculate the
+loss:
+
+.. doctest::
+
+    model = LeNet5()
+
+    # Input data and label
+    x = np.random.rand(32, 1, 28, 28).astype(np.float32)
+    t = np.random.randint(0, 10, size=(32,)).astype(np.int32)
+
+    # Forward computation
+    y = model(x)
+
+    # Loss calculation
+    loss = F.softmax_cross_entropy(y, t)
+
+This is a primitive way to calculate a loss value from the output of the model.
+On the other hand, the loss computation can be included in the model itself by
+wrapping the model object (:class:`~chainer.Chain` or
+:class:`~chainer.ChainList` object) with a class inherited from
+:class:`~chainer.Chain`. The outer :class:`~chainer.Chain` should take the
+model defined above and register it through the constructor of its superclass
+or :meth:`~chainer.Chain.add_link`. :class:`~chainer.Chain` is actually
+inherited from :class:`~chainer.Link`, so that :class:`~chainer.Chain` itself
+can also be registedred as a trainable :class:`~chainer.Link` to another
+:class:`~chainer.Chain`. Actually, :class:`~chainer.links.Classifier` class to
+wrap the model and add the loss computation to the model already exists.
+Using this, the loss computation can be implanted to the model itself by this
+way:
+
+.. doctest::
+
+    model = L.Classifier(LeNet5())
+
+    # Foward & Loss calculation
+    loss = model(x, t)
+
+This class takes a model object as an iput argument and registers it to
+``predictor`` property as a trained parameter. Then calling the returned object
+as a function like the above code with feeding ``x`` and ``t`` as the input
+arguments, yield a loss value calculated from those two input variables.
+
+See the detailed implementation of :class:`~chainer.links.Classifier` from
+here: :class:`chainer.links.Classifier` and check the implementation by looking
+at the source.
+
+Chainer is flexible, so we can write our original network in many different
+ways. It would give users an intuitive way to design brand-new and complex
```

Comment:
From the above examples, we can see that Chainer provides the flexibility to write our original network in many different ways. Such flexibility intends to make it intuitive for users to design new and complex models.